01-cython-numba/report/02-description-of-tools.tex

```diff
@@ -1,5 +1,39 @@
 \section{Opis użytych narzędzi}
-
+Istnieje wiele narzędzi pozwalających na wykonywanie obliczeń w bardziej wydajny sposób w Pythonie. Wybraliśmy dwa z nich, Cythona i Numbę. Nasza decyzja motywowana była popularnością pakietów, jak również znajomość języka C używanego w Cythonie i łatwość obsługi obu bibliotek.
+        
 \subsection{Cython}
+Cython jest nadzbiorem Pythona, który dodatkowo umożliwia wykorzystanie funkcji napisanych w języku C jak również jego typów. To pozwala kompilatorowi na tworzenie bardzo wydajnego kodu w języku C ze źródłowego kodu Cythona. Jest on generowany raz a następnie kompilowany przy użyciu wszystkich bardziej popularnych wersji kompilatora C/C++.
+
+Głównym zastosowaniem Cythona jest optymalizacja złożonych obliczeniowo fragmentów kodu poprzez kompilację ich do języka C lub w całości zastąpieniem natywnym kodem. Niestety dodatkowe ograniczenia narzucane przez statyczne typowanie istotnie komplikują interakcję z bibliotekami napisanymi w Pythonie i często trzeba odpowiednio je integrować z kodem Cythona.
+
+Za:
+\begin{itemize}
+    \item Łatwość użycia
+    \item Wysoka wydajność
+    \item Możliwość używania samych dekoratorów jak i pełnego wykorzystania języka C
+\end{itemize}
+
+Przeciw:
+\begin{itemize}
+    \item Integracja kodu Cythona z Pythonowym może być istotnie utrudniona, zwłaszcza podczas używania zewnętrznych bibliotek
+    \item Wymaga zainstalowanego dodatkowo kompilatora C/C++
+\end{itemize}
+
 
 \subsection{Numba}
+Numba - JIT kompilator Pythona. Pozwala na optymalizację i kompilację  kodu Pythona do kodu maszynowego.
+
+Numba daje najlepsze wyniki przy zastosowaniu jej na zwykłych pętlach oraz kodzie napisanym przy użyciu NumPy’a. Nie wspiera Pandasa. Ma problemy ze zwykłymi funkcjami napisanymi w Pythonie, które nie są typowane statycznie (to jest do sprawdzenia jeszcze). Dobrze współpracuje ze Sparkiem, Daskiem i Jupyterem.
```

Comment:
`Ma problemy ze zwykłymi funkcjami, wywołującymi inne funkcje, które nie mają określonych typów argumentów.` zamiast `Ma problemy ze zwykłymi funkcjami napisanymi w Pythonie, które nie są typowane statycznie (to jest do sprawdzenia jeszcze)`